examples/libxml2/libxml2_example.cc

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "examples/libxml2/xml_writer.h"
+#include "libxml/parser.h"
+#include "src/libfuzzer_protobuf_mutator.h"
+#include "xml.pb.h"  // NOLINT
+
+using examples::libxml2::Xml;
+
+namespace {
+google::protobuf::LogSilencer log_silincer;
+}
+
+extern "C" size_t LLVMFuzzerCustomMutator(uint8_t* data, size_t size,
+                                          size_t max_size, unsigned int seed) {
+  Xml message;
+  // Fuzzer can consume plane XML. In this case it will fail to parse it as
+  // proto.
+  bool is_proto = protobuf_mutator::ParseTextMessage(data, size, &message);
+  // If it plain XML or just time to tome we wrap XML as content of proto.
```

Comment:
Moved some code into src/xml.
unfortunately there is options field which is libxml2 specific.
But I guess it cant be solved with extensions, which not yet supported by mutator. It should not be hard.